app/async_database.py

```diff
@@ -0,0 +1,165 @@
+import aiosqlite
+import discord
+from database import UserProfile
+
+
+class AsyncDatabase:
+    """Class to store user profile and channel data asynchronously."""
+
+    db_path = "demo.db"
+
+    async def setup(self) -> None:
+        """Init for all asynchronous work for database."""
+        async with aiosqlite.connect(self.db_path) as db:
+            # initialize tables
+            await db.execute("""CREATE TABLE IF NOT EXISTS Guilds (
+                                id int,
+                                channel int,
+                                enabled bool)""")
+            await db.commit()
+            await db.execute("""CREATE TABLE IF NOT EXISTS Users (
+                                user_id int,
+                                guild_id int,
+                                cps float,
+                                coins int)""")
+            await db.commit()
+
+    async def add_guild(self, guild: discord.Guild) -> None:
+        """Add a guild to the database.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild that the user is in
+
+        """
+        async with aiosqlite.connect(self.db_path) as db:
+            await db.execute(
+                "INSERT INTO Guilds(id, channel, enabled) VALUES (?,?,?)",
```

Comment:
Guilds should have multiple channels be enabled: this shouldn't be necessary. When enabling a channel, just insert the guild and the channel id. When disabling it, remove them.